src/extra/utils/functions.py

```diff
@@ -23,6 +23,53 @@ def gaussian(x, y0, A, μ, σ):
     return y0 + (A / (σ * np.sqrt(2*np.pi))) * np.exp(-(x - μ)**2 / (2 * σ**2))
 
 
+def fit_gaussian(ydata, return_all=False, xdata=None, p0=None, **kwargs):
+    """Fit a Gaussian to some data.
+
+    This uses [curve_fit()][scipy.optimize.curve_fit] to fit a Gaussian (from
+    [gaussian()][extra.utils.gaussian]) to `ydata`. If `p0` is not passed the
+    function will set them to reasonable defaults. It will return `None` if
+    fitting fails, or if there are no non-NaN values in `ydata`.
+
+    Args:
+        ydata (array_like): The data to fit. NaN's will automatically be masked
+            before fitting.
+        return_all (bool): Whether to return everything from
+            [curve_fit()][scipy.optimize.curve_fit], or just `popt`.
+        p0 (list): A list of `[y0, A, μ, σ]` to match the arguments to
+            [gaussian()][extra.utils.gaussian].
+        **kwargs (): All other keyword arguments will be passed to
+            [curve_fit()][scipy.optimize.curve_fit].
+    """
+    if xdata is None:
+        xdata = np.arange(len(ydata))
+
+    from scipy.optimize import curve_fit
+
+    # Filter nans
+    nan_mask = ~np.isnan(ydata)
+    xdata = xdata[nan_mask]
+    ydata = ydata[nan_mask]
```

Comment:
It looks like recent versions of scipy have a `nan_policy` parameter, where `'omit'` is one option. However our 202401 environment still has scipy 1.8, so that's a question for another day.